### PR TITLE
feat: Job overview page

### PR DESCRIPTION
**Pull request by ShashwatDadhich**
# Description

The PR contains a new feature of job overview page.
Figma link = [figma](https://www.figma.com/file/83f7v1xGP6sYQC9XFHlqc7/Jobs?node-id=360%3A2212&t=vi4c8MXhGzZ8AMu9-0)

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

It has been tested locally by making changes and seeing it been reflected.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/Overview/Overview.scss [hunk 1]
```diff
@@ -36,4 +36,20 @@
     .modal__body {
         width: 600px;
     }
+}
+
+.ci-pipeline-name-color{
+    color: var(--B500);
```
Comment (sohel-rp) [inline src/components/app/Overview/Overview.scss:42]: use cb-5 class

File: src/components/app/Overview/Overview.scss [hunk 2]
```diff
@@ -36,4 +36,20 @@
...
+}
+
+.icon{
+    position: absolute;
+left: 45.83%;
+right: 45.83%;
+top: 29.17%;
+bottom: 29.17%;
```
Comment (sohel-rp) [inline src/components/app/Overview/Overview.scss:50]: why are we doing this?

File: src/components/app/Overview/Overview.scss [hunk 3]
```diff
@@ -36,4 +36,20 @@
...
+
+/* N0 */
+
+background: #FFFFFF;
```
Comment (sohel-rp) [inline src/components/app/Overview/Overview.scss:54]: use bcn-0 class

File: src/css/base.scss [hunk 4]
```diff
@@ -2685,6 +2690,17 @@ textarea::placeholder {
...
 
+.dc__description-textarea {
+    width: 100% !important;
+    min-height: 100px;
+    padding: 8px;
+    border: 1px solid #ccc;
+    border-radius: 4px;
+    font-size: 14px;
+    line-height: 20px;
+    color: #333;
```
Comment (sohel-rp) [inline src/css/base.scss:2701]: Is this the right color?

File: src/services/service.ts [hunk 5]
```diff
@@ -150,6 +150,10 @@ export function getAppOtherEnvironment(appId): Promise<AppOtherEnvironment> {
     return get(URL);
 }
 
+export function getJobCIPiprline(jobId){
```
Comment (sohel-rp) [inline src/services/service.ts:153]: Fix typo

File: src/components/app/Overview/Overview.tsx [hunk 6]
```diff
@@ -1,50 +1,77 @@
...
+    const [editMode, setEditMode] = useState(false)
+    const [newDescription, setNewDescription] = useState<string>(appMetaInfo?.description)
     const [externalLinksAndTools, setExternalLinksAndTools] = useState<ExternalLinksAndToolsType>({
         fetchingExternalLinks: true,
         externalLinks: [],
         monitoringTools: [],
     })
-    const [otherEnvsLoading, otherEnvsResult] = useAsync(() => Promise.all([getAppOtherEnvironment(appId), getModuleInfo(ModuleNameMap.ARGO_CD)]), [appId])
+    const [otherEnvsLoading, otherEnvsResult] = useAsync(
+        () => Promise.all([getAppOtherEnvironment(appId), getModuleInfo(ModuleNameMap.ARGO_CD)]),
```
Comment (sohel-rp) [inline src/components/app/Overview/Overview.tsx:53]: environment details are unnecessary for Jobs overview

File: src/components/app/Overview/Overview.tsx [hunk 7]
```diff
@@ -1,50 +1,77 @@
...
+        [appId],
+    )
     const isAgroInstalled: boolean = otherEnvsResult?.[1].result.status === ModuleStatus.INSTALLED
+    const [jobPipelines, setJobPipelines] = useState<JobPipeline[]>([])
+
+    useEffect(() => {
+        getJobCIPipeline(appId)
```
Comment (sohel-rp) [inline src/components/app/Overview/Overview.tsx:60]: try to add check in above useAsync as discussed

File: src/components/app/Overview/Overview.tsx [hunk 8]
```diff
@@ -115,15 +143,34 @@ export default function AppOverview({ appMetaInfo, getAppMetaInfoRes }: AppOverv
...
+                description: newDescription,
+            }
+
+            const appLabel = await createAppLabels(payload)
+
+            setNewDescription(appLabel.result.description)
+
+            setEditMode(false)
+        } catch (error) {
+            console.error(error)
```
Comment (sohel-rp) [inline src/components/app/Overview/Overview.tsx:164]: remove console error

File: src/components/app/Overview/Overview.tsx [hunk 9]
```diff
@@ -157,7 +204,7 @@ export default function AppOverview({ appMetaInfo, getAppMetaInfoRes }: AppOverv
                     </div>
                     <div className="flex fs-12 fw-4 lh-16 cn-7 cursor" onClick={toggleTagsUpdateModal}>
                         <EditIcon className="icon-dim-16 scn-7 mr-4" />
-                        Edit tags
+                        Edit
```
Comment (sohel-rp) [inline src/components/app/Overview/Overview.tsx:207]: is this expected?

File: src/components/app/Overview/Overview.tsx [hunk 10]
```diff
@@ -280,25 +327,166 @@ export default function AppOverview({ appMetaInfo, getAppMetaInfoRes }: AppOverv
...
-                {otherEnvsLoading ? (
-                    <div className="dc__loading-dots" />
-                ) : renderDeploymentComponent()}
+                {otherEnvsLoading ? <div className="dc__loading-dots" /> : renderDeploymentComponent()}
+            </div>
+        )
+    }
+
+    const renderWorkflowsStatus = () => {
+        const renderWorkflowComponent = () => {
```
Comment (sohel-rp) [inline src/components/app/Overview/Overview.tsx:336]: Don't write functions that return JSX element inside other

File: src/components/app/Overview/Overview.scss (hunk 1 above)
Comment (ShashwatDevtron) [reply src/components/app/Overview/Overview.scss:42]: done

Comment (ShashwatDadhich) [reply src/components/app/Overview/Overview.scss:42]: Done

File: src/components/app/Overview/Overview.scss (hunk 3 above)
Comment (ShashwatDadhich) [reply src/components/app/Overview/Overview.scss:54]: done

File: src/components/app/Overview/Overview.tsx (hunk 6 above)
Comment (ShashwatDadhich) [reply src/components/app/Overview/Overview.tsx:53]: Yes but we are using the same page for apps to render deployment status

File: src/components/app/Overview/Overview.tsx (hunk 9 above)
Comment (ShashwatDadhich) [reply src/components/app/Overview/Overview.tsx:207]: Yes

File: src/components/app/Overview/Overview.tsx (hunk 10 above)
Comment (ShashwatDadhich) [reply src/components/app/Overview/Overview.tsx:336]: done

File: src/css/base.scss (hunk 4 above)
Comment (ShashwatDadhich) [reply src/css/base.scss:2701]: Yes checked it

File: src/components/app/Overview/Overview.tsx [hunk 11]
```diff
@@ -0,0 +1,515 @@
...
+                    <RocketIcon className="icon-dim-20 scn-9 mr-8" />
+                    Deployments
+                </div>
+                {otherEnvsLoading ? <div className="dc__loading-dots" /> : renderDeploymentComponent()}
+            </div>
+        )
+    }
+
+    const renderWorkflowComponent = () => {
+        if (jobPipelines != null) {
```
Comment (sohel-rp) [inline src/components/app/Overview/Overview.tsx:338]: change this check to `Array.isArray(jobPipelines) && jobPipelines.length > 0`

File: src/components/app/Overview/Overview.tsx [hunk 12]
```diff
@@ -0,0 +1,515 @@
...
+                            <button className="btn btn-link p-0 fw-6 cb-5" type="submit" onClick={handleSave}>
+                                Save
+                            </button>
+                        </div>
+                    ) : (
+                        <div
+                            className="flex fs-12 fw-4 lh-16 cn-7 cursor ml-auto"
+                            onClick={() => {
+                                setEditMode(true)
+                            }}
```
Comment (sohel-rp) [inline src/components/app/Overview/Overview.tsx:457]: extract this function

**Pull request merged**